File: practice/mathematics/fundamentals/connecting_towns/ConnectingTowns.java
import java.util.Scanner;

public class ConnectingTowns {
    public static void main(String[] args) {
        Scanner stdin = new Scanner(System.in);
        int cases = stdin.nextInt();
        for(int i = 0; i < cases; i++) {
            int l = stdin.nextInt();
            int routes = 1;
            for(int j = 0; j < l - 1; j++) {
                routes *= stdin.nextInt();
                routes %= 1234567;
            }
            System.out.println(routes);
        }
        stdin.close();
    }
}

File: practice/mathematics/fundamentals/connecting_towns/ConnectingTowns.scala
import scala.io.Source

object ConnectingTowns extends App {
  val lines = Source.stdin.getLines().drop(1)
  val routesArray = lines.toList.filter(_.contains(" "))
  val routes = routesArray.map(_.split(" ").map(_.toLong).foldLeft(1L)(_ * _ % 1234567))
  println(routes.mkString("\n"))
}
